docs/topics/whatsnew18.md

```diff
@@ -223,6 +223,53 @@ Here are some of the key differences between the two layouts:
 | main  | src/main/AndroidManifest.xml  | src/<b>android</b>Main/AndroidManifest.xml  |
 | debug | src/debug/AndroidManifest.xml | src/<b>android</b>Debug/AndroidManifest.xml |
 
+#### The relation between Android tests and common tests
+
+The new Android source set layout changes the relation between Android-instrumented tests (renamed to `androidInstrumentedTest` in the new layout)
+and common tests.
+
+Previously, there was a default `dependsOn` relation between `androidAndroidTest` and `commonTest`. In practice, it meant the following:
+
+* The code in `commonTest` was available in `androidAndroidTest`.
+* `expect` declarations in `commonTest` had to have corresponding `actual` realizations in `androidAndroidTest`.
+* Tests declared in `commonTest` were running as Android instrumented tests.
+
+In the new Android source set layout, the `dependsOn` relation is not added by default. If you prefer the previous behavior,
+manually declare this relation:
+
+```kotlin
+kotlin {
+// ...
+    sourceSets {
+        val commonTest by getting
+        val androidInstrumentedTest by getting {
+            dependsOn(commonTest)
+        }
+    }
+}
+```
+
+#### Support for Android flavors
+
+The Android Gradle plugin creates a handful of source sets based on `buildType`. Along with the default `debug` and `release`,
```

Comment:
I think the people for whom this part is relevant are well-familiar with Android source sets, so we might cut the introduction a bit:

> Previously, source sets that corresponded to Android Source Sets with `BuildType` (`debug`/`release`) or custom flavors (e.g. `demo`/`full`) were created eagerly by Kotlin Gradle Plugin and thus accessible by constructions liks `val androidDebug by getting { ... }`
> Now they are created in the `afterEvaluate` phase. It makes such expressions ...